types/single-spa-react.d.ts

```diff
@@ -1,21 +1,31 @@
 import * as React from "react";
-import * as ReactDOM from "react-dom";
 import { AppProps, CustomProps, LifeCycleFn } from "single-spa";
 
 export const SingleSpaContext: React.Context<CustomProps & AppProps>;
 
+type DeprecatedRenderTypes =
+  | "createBlockingRoot"
+  | "unstable_createRoot"
+  | "unstable_createBlockingRoot";
+
+type LegacyRenderType = "hydrate" | "render";
+
+type RenderType =
+  // React 18
+  "createRoot" | "hydrateRoot" | LegacyRenderType;
+
 export interface SingleSpaReactOpts<RootComponentProps> {
-  React: typeof React;
-  ReactDOM: typeof ReactDOM;
+  React: any;
```

Comment:
Whenever possible, we should avoid the `any` type. I'm not sure what was wrong with the previously used type `typeof React` since React is a peer dependency. If I'm understanding peer dependencies correctly (and I'm not really sure if I am), the type of that import would be defined by the version of React that the user has.

And the same point for the other props.

So I would expect the types to be:
```ts
import * as ReactDOM from "react-dom";
import * as ReactDOMClient from "react-dom/client";
//...
export interface SingleSpaReactOpts<RootComponentProps> {
  React: typeof React;
  ReactDOM?: typeof ReactDOM;
  ReactDOMClient?: typeof ReactDOMClient;
  //...
}
```